init.ps1

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env pwsh
+
+Write-Host "`nStarting project setup..." -ForegroundColor Cyan
+
+function Run-Command($cmd, $errorMsg) {
+    try {
+        Write-Host "Running: $cmd" -ForegroundColor Yellow
+        Invoke-Expression $cmd
+    } catch {
+        Write-Host "$errorMsg" -ForegroundColor Red
+        exit 1
+    }
+}
+
+# 1. Install vsts-npm-auth globally
+Run-Command "npm install -g vsts-npm-auth" "Failed to install vsts-npm-auth."
+
+# 2. Authenticate with vsts-npm-auth
+if (Test-Path ".npmrc") {
```

Comment:
Consider wrapping the body the following: 

```pwsh
Push-Location $PSScriptRoot
try {

}
finally {
  Pop-Location
}
```

Lacking that if `init.ps1` is run from a diff directory calls like `Test-Path` here won't do what you're expecting.